modules/mobile/app/controllers/mobile/v0/facilities_info_controller.rb

```diff
@@ -27,27 +32,47 @@ def sort(facilities, sort_method)
         when 'home', 'current'
           facilities.sort_by(&:miles)
         when 'alphabetical'
-          facilities.sort_by(&:name)
+          sort_by_name(facilities)
         when 'appointments'
-          sort_by_recent_appointment(facilities)
+          sort_by_recent_appointment(sort_by_name(facilities))
         else
           raise Common::Exceptions::ValidationErrorsBadRequest.new(
-            detail: 'Invalid sort method', source: self.class.to_s, sort_method: sort_method
+            detail: 'Sort method not implemented', source: self.class.to_s, sort_method: sort_method
           )
         end
       end
 
+      def sort_by_name(facilities)
+        facilities.sort_by(&:name)
+      end
+
       def sort_by_recent_appointment(facilities)
-        facilities.sort_by(&:name) # entries not in recent appointments will be sorted alphabetically
-        appointments = Mobile::V0::Appointment.get_cached(@current_user).sort_by(&:start_date_utc)
-        if appointments.blank?
-          raise Common::Exceptions::RecordNotFound.new(
-            detail: 'Could not fetch user appointments', source: self.class.to_s
-          )
+        appointments = Mobile::V0::Appointment.get_cached(@current_user)&.sort_by(&:start_date_utc)
+
+        log_nil_cache if appointments.nil?
+        return facilities if appointments.blank?
+
+        appointment_facility_ids = appointments.map(&:facility_id).uniq
+
+        appointment_facility_ids.map! do |facility_id|
+          Mobile::V0::Appointment.convert_to_non_prod_id!(facility_id)
         end
-        recent_facilities = appointments.map(&:facility_id).uniq
-        rf_hash = recent_facilities.each_with_index.to_h
-        facilities.sort_by { |facility| [rf_hash[facility.id] || recent_facilities.size, facility.id] }
+
+        appointment_facilities_hash = appointment_facility_ids.each_with_index.to_h
+
+        # appointment_facility_ids.size ensures any facility not found in appointment_facilities_hash is pushed to the
+        # bottom of the array
+        facilities.sort_by { |facility| appointment_facilities_hash[facility.id] || appointment_facility_ids.size }
```

Comment:
This was able to be simplified because we want to leave any facility NOT in `recent_facilities` untouched, as they are already alphabetized from earlier. Now this will just push all non-recent facilities to the bottom

Reply:
If the original was sorting by id that was a mistake. The facilities that don't match past appointments should be sorted alphabetically after the ones that do match.